engine/pkg/orm/model/logic_epoch.go

```diff
@@ -16,63 +16,107 @@ package model
 import (
 	"context"
 
+	"github.com/pingcap/tiflow/pkg/errors"
 	"gorm.io/gorm"
 	"gorm.io/gorm/clause"
 )
 
 const (
-	defaultEpochPK  = 1
 	defaultMinEpoch = 1
 )
 
 // LogicEpoch is used to generate increasing epoch
 type LogicEpoch struct {
 	Model
-	Epoch int64 `gorm:"type:bigint not null default 1"`
+	JobID string `gorm:"type:varchar(64) not null;uniqueIndex:uidx_jk"`
+	Epoch int64  `gorm:"type:bigint not null default 1"`
 }
 
-// InitializeEpoch insert the only record into the backend table `logic_epoches`
-func InitializeEpoch(ctx context.Context, db *gorm.DB) error {
+// EpochClient defines the client to generate epoch
+type EpochClient interface {
+	// Initialize inserts a record into the backend table `logic_epoches`
+	Initialize(ctx context.Context) error
+
+	// GenEpoch increases the backend epoch by 1 and return the new epoch
+	GenEpoch(ctx context.Context) (int64, error)
+
+	// Close releases some inner resources
+	Close() error
+}
+
+// NewEpochClient news a EpochClient
+func NewEpochClient(jobID string, db *gorm.DB) *epochClient {
+	return &epochClient{
+		jobID: jobID,
+		db:    db,
+	}
+}
+
+type epochClient struct {
+	jobID string
+	db    *gorm.DB
+}
+
+// Initialize implements Initialize of EpochClient
+func (e *epochClient) Initialize(ctx context.Context) error {
+	if e.db == nil {
+		return errors.ErrMetaParamsInvalid.GenWithStackByArgs("inner db is nil")
+	}
+
 	// Do nothing on conflict
-	// INSERT INTO `logic_epoches` (`created_at`,`updated_at`,`epoch`,`seq_id`) VALUES
-	// ('2022-05-04 14:02:08.624','2022-05-04 14:02:08.624',1,1) ON DUPLICATE KEY UPDATE `seq_id`=`seq_id`
-	return db.WithContext(ctx).
+	return e.db.WithContext(ctx).
 		Clauses(clause.OnConflict{DoNothing: true}).
 		Create(&LogicEpoch{
-			Model: Model{
-				SeqID: defaultEpochPK,
-			},
+			JobID: e.jobID,
 			Epoch: defaultMinEpoch,
 		}).Error
 }
 
-// GenEpoch will increasing the backend epoch by 1 and return the new epoch
-func GenEpoch(ctx context.Context, db *gorm.DB) (int64, error) {
+// GenEpoch implements GenEpoch of EpochClient
+func (e *epochClient) GenEpoch(ctx context.Context) (int64, error) {
 	var epoch int64
-	err := db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
-		//(1)update epoch = epoch + 1
-		if err := tx.Model(&LogicEpoch{
-			Model: Model{
-				SeqID: defaultEpochPK,
-			},
-		}).Update("epoch", gorm.Expr("epoch + ?", 1)).Error; err != nil {
-			// return any error will rollback
-			return err
-		}
-
-		//(2)select epoch
-		var logicEp LogicEpoch
-		if err := tx.First(&logicEp, defaultEpochPK).Error; err != nil {
-			return err
-		}
-		epoch = logicEp.Epoch
-
-		// return nil will commit the whole transaction
-		return nil
-	})
+	err := e.db.Where("job_id = ?", e.jobID).
```

Comment:
Why do we need `Where("job_id = ?", e.jobID)`?

Reply:
Different job has its own epoch, or we may have heavy row lock.